tools/generate_property_translations.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+#
+# Generates translation strings from gecko property files for lupdate
+# to process
+
+import os, os.path, sys, re, dataclasses, functools
+
+PREFIX = "sailfish_components_webview_popups-la"
+
+class Generator:
+    RE = re.compile(r"(?P<name>[\w-]+)\s*=\s*(?P<text>.*)\s*")
+    SPLITTER = re.compile(r"([A-Za-z0-9][a-z0-9]*)")
+    PLACEHOLDER = re.compile(r"%(?:([0-9]+)\$)?S")
+
+    def __init__(self, file):
+        self.file = file
+
+    @property
+    @functools.cache
+    def name(self):
+        """Basename from self.file without file extension and
+        with dashes replaced by underscores"""
+        return os.path.splitext(os.path.basename(self.file))[0].replace("-", "_")
+
+    @staticmethod
+    def _convert_key(name):
+        """Convert from camelCase to snake_case"""
+        return "_".join(Generator.SPLITTER.findall(name)).lower()
+
+    @staticmethod
+    def _convert_text(text):
+        """Replace gecko's placeholders with Qt's and
+        remove keyboard shortcut marks"""
```

Comment:
I guess hints would be more correct than marks.

Reply:
Yeap.